xarray/tests/duckarrays/test_sparse.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import pytest
+
+from xarray import DataArray, Dataset, Variable
+
+# isort: off
+# needs to stay here to avoid ImportError for the strategy imports
+pytest.importorskip("hypothesis")
+# isort: on
+
+from .. import assert_allclose
+from . import base
+from .base import strategies
+
+sparse = pytest.importorskip("sparse")
+
+
+@pytest.fixture(autouse=True)
+def disable_bottleneck():
+    from xarray import set_options
+
+    with set_options(use_bottleneck=False):
+        yield
+
+
+def create(op, shape, dtypes):
```

Comment:
Could we change this signature to

```suggestion
def create(shape, dtypes, op=None):
```
?

Presumably this creation function is to be called for many types of tests (not just reductions), and some of those tests won't require a specify operation argument. For example I don't need creation to depend on `op` if I'm just testing `Variable.__init__`.

(Also none of your reduction testing code seems to use this argument at the moment anyway?)

Reply:
`op` is used for `pint` where the unit has to be `"unitless"` in order for `cumprod` to work. However, I think I could just change `check_reduce` to verify that `cumprod` indeed raises an error and remove the parameter to `create` entirely.